### PR TITLE
Separate execution known issues

### LINKED ISSUE #370: Separate "execution" known issues
We need to categorize and separate "execution known issues" from the stable execution tests similar to the "compilation known issues". 
[AB#10625](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/10625)

### PR DESCRIPTION
**Pull request by zengin**
- `KnownFailuresRequested` was not a representative naming, especially after introducing new known issue type for execution tests. Instead I defined a new `TestType` enum separating stable and known issue tests for both compilation and execution:

```csharp
    public enum TestType
    {
        CompilationStable,
        CompilationKnownIssues,
        ExecutionStable,
        ExecutionKnownIssues
    }
```

This allows us a two layered approach to known issues. We can now add execution known issues into the code for further categorization.

- Fixed pipeline for weekly execution tests to include both stable tests and known issues to continue having the full suite in the dashboard.
- Created "execution known issues" pipelines for V1 and Beta to run them as part of PRs.
- Fixes #370

### CODE REVIEW

File: CsharpBetaExecutionKnownFailureTests/SnippetExecutionBetaKnownFailureTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using NUnit.Framework;
+
+using TestsCommon;
+
+namespace CsharpBetaExecutionKnownFailureTests
+{
+    [TestFixture]
+    public class SnippetExecutionBetaKnownFailureTests
+    {
+        private IConfidentialClientApplication _confidentialClientApp;
```
Comment (baywet) [inline CsharpBetaExecutionKnownFailureTests/SnippetExecutionBetaKnownFailureTests.cs:15]: nit: readonly?

File: TestsCommon/TestDataGenerator.cs [hunk 2]
```diff
@@ -559,8 +590,12 @@ public static IEnumerable<TestCaseData> GetExecutionTestData(RunSettings runSett
...
                    let fullPath = Path.Join(GraphDocsDirectory.GetSnippetsDirectory(testData.Version, runSettings.Language), testData.FileName)
                    let fileContent = File.ReadAllText(fullPath)
-                   let executionTestData = new ExecutionTestData(testData, fileContent)
-                   where !testData.IsKnownIssue // select compiling tests
+                   let executionTestData = new ExecutionTestData(testData with
+                   {
+                       TestName = testData.TestName.Replace("-compiles", "-executes")
+                   }, fileContent)
+                   where !testData.IsCompilationKnownIssue // select compiling tests
+                   && !(testData.IsExecutionKnownIssue ^ runSettings.TestType == TestType.ExecutionKnownIssues) // select known execution issues iff requested
```
Comment (baywet) [inline TestsCommon/TestDataGenerator.cs:598]: ooooh a XOR, I haven't seen those in a while!!

**Review by baywet — APPROVED** [review]

File: CsharpBetaExecutionKnownFailureTests/SnippetExecutionBetaKnownFailureTests.cs (hunk 1 above)
Comment (georgend) [reply CsharpBetaExecutionKnownFailureTests/SnippetExecutionBetaKnownFailureTests.cs:15]: Wouldn't readonly require it to be set in a constructor?

Comment (baywet) [reply CsharpBetaExecutionKnownFailureTests/SnippetExecutionBetaKnownFailureTests.cs:15]: right, and the method is not the constructor, sorry, tea not fully absorbed yet.

**Review by georgend — APPROVED** [review]

**Pull request merged**